src/main/java/org/rutebanken/tiamat/exporter/StreamingPublicationDelivery.java

```diff
@@ -248,6 +295,113 @@ private void prepareStopPlaces(ExportParams exportParams, Set<Long> stopPlacePri
         }
     }
 
+    private void prepareScheduledStopPoints(ExportParams exportParams, Set<Long> stopPlacePrimaryIds, org.rutebanken.netex.model.ServiceFrame netexServiceFrame) {
+        if (exportParams.getServiceFrameExportMode() == ExportParams.ExportMode.ALL && !stopPlacePrimaryIds.isEmpty()) {
+            logger.info("There are stop places to export");
+
+            final Iterator<org.rutebanken.tiamat.model.StopPlace> stopPlaceIterator = stopPlaceRepository.scrollStopPlaces(stopPlacePrimaryIds);
+
+            List<ScheduledStopPoint> netexScheduledStopPoints = new ArrayList<>();
+
+            List<JAXBElement<? extends StopAssignment_VersionStructure>> stopAssignment = new ArrayList<>();
+
+
+            while (stopPlaceIterator.hasNext()) {
+                final org.rutebanken.tiamat.model.StopPlace stopPlace = stopPlaceIterator.next();
+                covertStopPlaceToScheduledStopPoint(netexScheduledStopPoints, stopAssignment, stopPlace);
+
+            }
+
+
+            if (!netexScheduledStopPoints.isEmpty()) {
+                final ScheduledStopPointsInFrame_RelStructure scheduledStopPointsInFrame_relStructure = new ScheduledStopPointsInFrame_RelStructure();
+                setField(ScheduledStopPointsInFrame_RelStructure.class, "scheduledStopPoint", scheduledStopPointsInFrame_relStructure, netexScheduledStopPoints);
+                netexServiceFrame.setScheduledStopPoints(scheduledStopPointsInFrame_relStructure);
+
+
+                StopAssignmentsInFrame_RelStructure stopAssignmentsInFrame_RelStructure = new StopAssignmentsInFrame_RelStructure();
+                setField(StopAssignmentsInFrame_RelStructure.class, "stopAssignment", stopAssignmentsInFrame_RelStructure, stopAssignment);
+                netexServiceFrame.setStopAssignments(stopAssignmentsInFrame_RelStructure);
+            }
+
+        }
+    }
+
+    private void covertStopPlaceToScheduledStopPoint(List<ScheduledStopPoint> scheduledStopPoints, List<JAXBElement<? extends StopAssignment_VersionStructure>> netexPassengerStopAssignment, org.rutebanken.tiamat.model.StopPlace stopPlace) {
+
+        // Add stop place
+        final String netexId = stopPlace.getNetexId();
+        String stopPlaceName = null;
+        if (stopPlace.getName() != null) {
+            stopPlaceName = stopPlace.getName().getValue();
+        }
+        final long version = stopPlace.getVersion();
+        var stopPlaceNetexId = netexId.split(":")[2];
+        var scheduledStopPointNetexId = "NSR:ScheduledStopPoint:S" + stopPlaceNetexId;
```

Comment:
Burde vel bruke netex.validPrefix - property-verdien i stedet for å hardkode NSR:?

Reply:
✅